clients/src/main/java/org/apache/kafka/common/requests/ProduceRequest.java

```diff
@@ -133,6 +144,25 @@ public Struct toStruct() {
     }
 
     @Override
+    public String toString() {
+        // Store it in a local variable to protect against concurrent updates
+        Map<TopicPartition, MemoryRecords> partitionRecords = this.partitionRecords;
+
+        StringBuilder bld = new StringBuilder();
+        bld.append("(type=ProduceRequest")
+                .append(", acks=").append(acks)
+                .append(", timeout=").append(timeout);
+
+        if (partitionRecords == null)
+            bld.append(", partitions=").append(Utils.mkString(partitions, ","));
+        else
+            bld.append(", partitionRecords=(").append(Utils.mkString(partitionRecords));
```

Comment:
Yeah, actually I agree with this. These messages have started to pollute some test logs.